src/robotFactory.js

```diff
@@ -1,10 +1,89 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, {
+    x = 0,
+    y = 0,
+  }, chipVersion) {
+    Object.assign(this, {
+      name,
+      weight,
+      coords: {
+        x,
+        y,
+      },
+      chipVersion,
+    });
```

Comment:
I think it would be better to assign each property separately:
```suggestion
    this.name = name;
    this.weight = weight 
    etc...
```